src/actions/websocket.ts

```diff
@@ -326,17 +326,26 @@ function handleEvent(msg: WebSocketMessage) {
 
 function handleNewPostEvent(msg: WebSocketMessage) {
     return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
         const post = JSON.parse(msg.data.post);
 
-        dispatch(handleNewPost(msg));
-        getProfilesAndStatusesForPosts([post], dispatch, getState);
+        if (getCurrentChannelId(state) === post.channel_id) {
```

Comment:
this should be in if statement cause you should have already incremented the post visibility if you already have the post.

Reply:
you should handle it in the app, instead of relying on this event when the websocket gets the post back.

Reasoning: you create a post in the client, WS is disconnected so you miss the broadcast event but you still should increase the visibility